rust/benchmarks/src/bin/tpch.rs

```diff
@@ -1003,12 +1004,11 @@ async fn execute_query(
     if debug {
         pretty::print_batches(&result)?;
     }
-    Ok(())
+    Ok(result)
 }
 
-async fn convert_tbl(opt: ConvertOpt) -> Result<()> {
+async fn convert_tbl(opt: ConvertOpt) -> Result<Vec<arrow::record_batch::RecordBatch>> {
```

Comment:
I would avoid making the type here the same (and return empty vec) to let it compile. You could change the code in main fn instead to not have it in the main expression?

Reply:
Ok. I have refactored the main fn to allow the different result signatures as it is definitely better. Thanks 👍